Instrument Rollups CRUD app with user action telemetry

cjcenizal opened the pull request:
Actions tracked:

* "Load app" action
* "Delete job" action (single, many)
* "Start job" action (single, many)
* "Stop job" action (single, many)
* "Open details panel" action
* "View summary tab" action
* "View terms tab" action
* "View histogram tab" action
* "View metrics tab" action
* "View JSON tab" action

Telemetry shape as reported by `http://localhost:5601/api/stats?extended=true`:

```
{  
  "user_actions":[
    {
      "key":"app_load",
      "value":1
    },
    {
      "key":"job_create",
      "value":5
    },
    {
      "key":"job_delete",
      "value":1
    },
    {
      "key":"job_delete_many",
      "value":1
    },
    {
      "key":"job_start",
      "value":1
    },
    {
      "key":"job_start_many",
      "value":1
    },
    {
      "key":"job_stop",
      "value":1
    },
    {
      "key":"job_stop_many",
      "value":1
    },
    {
      "key":"show_details_click",
      "value":1
    },
    {
      "key":"detail_panel_summary_tab_click",
      "value":1
    },
    {
      "key":"detail_panel_terms_tab_click",
      "value":1
    },
    {
      "key":"detail_panel_histogram_tab_click",
      "value":1
    },
    {
      "key":"detail_panel_metrics_tab_click",
      "value":1
    },
    {
      "key":"detail_panel_json_tab_click",
      "value":2
    }
  ]
}
```

elasticmachine commented:
Pinging @elastic/es-ui

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3796/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3810/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3818/)

sebelga commented on x-pack/plugins/rollup/public/crud_app/sections/job_list/detail_panel/detail_panel.js:
```diff
@@ -96,7 +119,7 @@ export class DetailPanelUi extends Component {
       renderedTabs.push(
         <EuiTab
-          onClick={() => openDetailPanel({ panelType: tab, jobId: id })}
+          onClick={() => this.handleTabClick(tab)}
```
I think I prefer how you handled it below in the job table

```
onClick={() => {
  trackUserAction(tabToUserActionMap[tab]); // with "tabToUserActionMap" declared in the upper scope
  openDetailPanel({ panelType: tab, jobId: id }); // as before
}}
```

This removes the need of a click handler and is consistent with other places.

cjcenizal replied:
I agree, done!

jen-huang commented:
Oh, I had one question that is not a blocker: when I do bulk actions, it increments +1 regardless of how many jobs the action was performed on. For example, if I delete two jobs at once, `job_delete` increments by 1 instead of 2. From a telemetry perspective, do you think this matters? Is there value in potentially tracking bulk actions vs single actions?

cjcenizal commented:
@jen-huang I updated the PR to differentiate between actions which effect individual vs many jobs. I also added an action to track when the user loads the Rollup Jobs app. Could you take another look please?

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4657/)

cjcenizal commented:
Retest

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4673/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4914/)
